p24/src/refs.rs

```diff
@@ -0,0 +1,71 @@
+// f1 returns first or second element of tuple depending on the flag.
+pub fn f1(pair: &mut (u32, u32), flag: bool) -> &mut u32 {
+    match flag {
+        true => &mut pair.0,
+        false => &mut pair.1,
+    }
+}
+
+// f2 returns n-th element of slice.
+pub fn f2(slice: &mut [u32], n: usize) -> &mut u32 {
+    &mut slice[n]
+}
+
+// f3 returns n-th element of slice from the end.
+pub fn f3(slice: &mut [u32], n: usize) -> &mut u32 {
+    let len = slice.len();
+    &mut slice[len - n - 1]
+}
+
+// f4 breaks up a slice into four equal parts and returns all the parts.
+pub fn f4(slice: &mut [u32]) -> (&mut [u32], &mut [u32], &mut [u32], &mut [u32]) {
```

Comment:
Note that this does not implement fair division into chunks. For a slice of length 7 you get slices with lengths 1,1,1, and 4.

Also note that in the problem description you have an immutable slice in the input argument.